File: java/src/main/java/com/android/util/ToastUtil.java
package com.android.util;

import android.content.Context;
import android.graphics.Color;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;
import com.android.base.BaseApplication;
import com.android.java.R;

/**
 * Created by xuzhb on 2019/10/22
 * Desc:自定义Toast
 */
public class ToastUtil {

    public static void showToast(CharSequence text) {
        showToast(text, BaseApplication.getInstance().getApplicationContext(), true, false);
    }

    public static void showToast(CharSequence text, Context context, boolean isCenter, boolean longToast) {
        int duration = Toast.LENGTH_SHORT;
        if (longToast) {
            duration = Toast.LENGTH_LONG;
        }
        Toast toast = Toast.makeText(context.getApplicationContext(), text, duration);
        View layout = LayoutInflater.from(context.getApplicationContext()).inflate(R.layout.layout_toast, null);
        TextView toastTv = layout.findViewById(R.id.toast_tv);
        toastTv.setText(text);
        layout.setBackground(DrawableUtil.createSolidShape(SizeUtil.dp2px(3), Color.parseColor("#CC1F2735")));
        if (isCenter) {
            toast.setGravity(Gravity.CENTER, 0, 0);
        }
        toast.setView(layout);
        toast.show();
    }

    public static void toast(CharSequence text) {
        Toast.makeText(BaseApplication.getInstance().getApplicationContext(), text, Toast.LENGTH_SHORT).show();
    }

    public static void longToast(CharSequence text) {
        Toast.makeText(BaseApplication.getInstance().getApplicationContext(), text, Toast.LENGTH_LONG).show();
    }

}

File: kotlin/src/main/java/com/android/util/ToastUtil.kt
package com.android.util

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.view.Gravity
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast
import com.android.base.BaseApplication
import com.android.basicproject.R

/**
 * Created by xuzhb on 2019/9/26
 * Desc:自定义Toast
 */
object ToastUtil {

    fun showToast(
        text: CharSequence,
        isCenter: Boolean = false,
        longToast: Boolean = false,
        context: Context = BaseApplication.instance
    ) {
        val duration = if (longToast) Toast.LENGTH_LONG else Toast.LENGTH_SHORT
        val toast = Toast.makeText(context.applicationContext, text, duration)
        val layout = LayoutInflater.from(context.applicationContext).inflate(R.layout.layout_toast, null)
        val toastTv: TextView = layout.findViewById(R.id.toast_tv)
        toastTv.setText(text)
        val toastShape = GradientDrawable()
        toastShape.setColor(Color.parseColor("#CC1F2735"))
        toastShape.cornerRadius = SizeUtil.dp2px(3f)
        layout.background = toastShape
        if (isCenter) {
            toast.setGravity(Gravity.CENTER, 0, 0)
        }
        toast.view = layout
        toast.show()
    }

    fun showToast(
        text: CharSequence,
        gravity: Int = Gravity.CENTER,
        xOffset: Int = 0,
        yOffset: Int = 0,
        longToast: Boolean = false,
        context: Context = BaseApplication.instance
    ) {
        val duration = if (longToast) Toast.LENGTH_LONG else Toast.LENGTH_SHORT
        val toast = Toast.makeText(context.applicationContext, text, duration)
        val layout = LayoutInflater.from(context.applicationContext).inflate(R.layout.layout_toast, null)
        val toastTv: TextView = layout.findViewById(R.id.toast_tv)
        toastTv.setText(text)
        val toastShape = GradientDrawable()
        toastShape.setColor(Color.parseColor("#CC1F2735"))
        toastShape.cornerRadius = SizeUtil.dp2px(3f)
        layout.background = toastShape
        toast.setGravity(gravity, xOffset, yOffset)
        toast.view = layout
        toast.show()
    }

    fun toast(text: CharSequence, context: Context = BaseApplication.instance) {
        Toast.makeText(context, text, Toast.LENGTH_SHORT).show()
    }

    fun longToast(text: CharSequence, context: Context = BaseApplication.instance) {
        Toast.makeText(context, text, Toast.LENGTH_LONG).show()
    }

}